rotkehlchen/assets/asset.py

```diff
@@ -1129,3 +1148,96 @@ def from_identifier(
             )
         except DeserializationError:
             return None
+
+
+# Create a generic variable that can be 'CustomAsset', or any subclass such as stock, painting, etc.  # noqa: E501
+R = TypeVar('R', bound='CustomAsset')
+
+
+@dataclass(init=True, repr=True, eq=False, order=False, unsafe_hash=False, frozen=True)
+class CustomAsset(Asset):
+    """Class to represent assets added by users that are not crypto coins"""
+    custom_asset_type: str = field(init=False)
+    notes: Optional[str] = field(init=False)
+
+    @classmethod
+    def initialize(  # type: ignore  # figure out a way to make mypy happy
+            cls: Type[R],
+            identifier: str,
+            name: str,
+            custom_asset_type: str,
+            notes: Optional[str] = None,
+            swapped_for: Optional[Asset] = None,
+            started: Optional[Timestamp] = None,
+    ) -> R:
+        """Initialize a custom asset from fields"""
+        custom_asset = cls('whatever', direct_field_initialization=True)
+        object.__setattr__(custom_asset, 'identifier', identifier)
+        object.__setattr__(custom_asset, 'name', name)
+        object.__setattr__(custom_asset, 'symbol', None)
+        object.__setattr__(custom_asset, 'asset_type', AssetType.CUSTOM_ASSET)
+        object.__setattr__(custom_asset, 'custom_asset_type', custom_asset_type)
+        object.__setattr__(custom_asset, 'notes', notes)
+        object.__setattr__(custom_asset, 'started', started)
+        object.__setattr__(custom_asset, 'forked', None)
+        object.__setattr__(custom_asset, 'swapped_for', swapped_for)
+        object.__setattr__(custom_asset, 'cryptocompare', None)
+        object.__setattr__(custom_asset, 'coingecko', None)
+        return custom_asset
+
+    def __post_init__(
+            self,
+            form_with_incomplete_data: bool = False,
+            direct_field_initialization: bool = False,
+    ) -> None:
+        if direct_field_initialization:
+            return
+
+        object.__setattr__(self, 'identifier', self.identifier)
+        super().__post_init__(form_with_incomplete_data)
+        # TODO: figure out a way to move this out. Moved in here due to cyclic imports
+        from rotkehlchen.assets.resolver import AssetResolver  # isort:skip  # noqa: E501  # pylint: disable=import-outside-toplevel
+
+        data = AssetResolver().get_asset_data(self.identifier)  # pylint: disable=no-member
+        object.__setattr__(self, 'custom_asset_type', data.custom_asset_type)
+        object.__setattr__(self, 'notes', data.notes)
+
+    @classmethod
+    def from_identifier(
+            cls: Type[R],
+            identifier: str,
+            form_with_incomplete_data: bool = True,
```

Comment:
What is the purpose of `form_with_incomplete_data` ? If I understand correctly, right now it doesn't make any sense / impact for custom assets

Reply:
It does not. It should not be here, unless it's here only for compatibility with other function's arguments.